src/containers/chat/list-of-users-with-search/ListOfUsersWithSearch.tsx

```diff
@@ -0,0 +1,80 @@
+import { FC, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import SimpleBar from 'simplebar-react'
+
+import Box from '@mui/material/Box'
+import Divider from '@mui/material/Divider'
+import Typography from '@mui/material/Typography'
+
+import ChatItem from '~/containers/chat/chat-item/ChatItem'
+import FilterInput from '~/components/filter-input/FilterInput'
+import { useAppSelector } from '~/hooks/use-redux'
+
+import { ChatResponse, SizeEnum } from '~/types'
+import { styles } from '~/containers/chat/list-of-users-with-search/ListOfUsersWithSearch.styles'
+import { filterChats } from './ListOfUsersWithSearch.constants'
+
+interface ListOfUsersWithSearchProps {
+  listOfChats: ChatResponse[]
+  isSelectedChat: string
+  setIsSelectedChat: (id: string) => void
+  closeDrawer?: () => void
+}
+
+const ListOfUsersWithSearch: FC<ListOfUsersWithSearchProps> = ({
+  listOfChats,
+  isSelectedChat,
+  setIsSelectedChat,
+  closeDrawer
+}) => {
+  const [search, setSearch] = useState<string>('')
+
+  const { userId } = useAppSelector((state) => state.appMain)
+
+  const { t } = useTranslation()
+
+  const chats = useMemo(() => {
+    const filteredChats = filterChats(listOfChats, userId, search)
+    return filteredChats.map((item: ChatResponse) => (
+      <ChatItem
+        closeDrawer={closeDrawer}
+        isSelectedChat={isSelectedChat}
```

Comment:
You should think about the way you handle this. Cause atm all your items would re-render after selecting chat. And after each other selection as well...

Reply:
To ensure proper functionality, I first verify that the chat is active. I then pass this information as a prop to the ChatItem component. Additionally, I have wrapped the ChatItem component in memo, resulting in correct operation. This approach only renders the selected component and the component that was chosen, while leaving other components without re-renders.